src/Workspaces/CoreTest/SymbolKeyTests.cs

```diff
@@ -623,19 +623,65 @@ void goo()
             var document = workspace.AddDocument(project.Id, "testdocument", sourceText);
 
             var firstModel = await document.GetSemanticModelAsync();
-            var tree1 = await document.GetSyntaxTreeAsync();
-            var basemethod1 = tree1.FindTokenOnLeftOfPosition(position, CancellationToken.None).GetAncestor<CSharp.Syntax.BaseMethodDeclarationSyntax>();
+
+            // Ensure we prime the reuse cache with the true semantic model.
```

Comment:
I'd be curious if we had some way to measure the impact of the requirement to "prime" this -- if the common callers of this are all completion, are they going to get a speculative model if the user opens the file and types their first key, or will this not have been primed yet?

Reply:
> are they going to get a speculative model if the user opens the file and types their first key, or will this not have been primed yet?

They will not get this.  Your first keystroke pays the price of getting the semantic model.  But from that point on, as long as you're in a method body, you get the speculative ones.  The point is to amortize the cost.  Not ensure that all the calls never have this cost.

Reply:
Will raise concern back on main thread.